lib/chef/resource/yum_package.rb

```diff
@@ -38,7 +38,7 @@ def arch(arg=nil)
         set_or_return(
           :arch,
           arg,
-          :kind_of => [ String ]
+          :kind_of => [ String, Array ]
```

Comment:
#3233 indicates that you are using 'package.arch' in a multi-package pattern. Is this an example of what used to not work and what would now work?

```
package [ "cups.x86_64", "vim" ]
```

This line makes me wonder if this does the right thing:

```
package [ "cups", "vim" ] do
  arch [ "x86_64", "i386" ]
end
```

The first syntax seems friendlier, but I'd expect this to work too.